scripts/inventory/getinventory.py

```diff
@@ -17,13 +35,68 @@ def read_inventory_file(filename):
 
 
 def get_instance_group(instance, group):
-    servers = read_inventory_file(os.path.join('fab', 'inventory', instance))
+    servers = read_inventory_file(
+        os.path.join(ROOT, 'fab', 'inventory', instance))
     return servers[group]
 
-if __name__ == '__main__':
-    import sys
-    instance = sys.argv[1]
-    group = sys.argv[2]
-    servers = get_instance_group(instance, group)
-    for server in servers:
-        print server
+
+def main():
+    prog = os.environ.get("SCRIPT", sys.argv[0])
+    parser = ArgParser(
+        prog=prog,
+        usage="{prog} [-h] environment group[:n] [{uprog}_ARGS]".format(
+            prog=prog,
+            uprog=prog.rsplit("/", 1)[-1].upper(),
+        ),
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+    parser.add_argument("environment",
+        help="Environment: production, staging, ...")
+    parser.add_argument("group",
+        help="Server group: postgresql, proxy, webworkers, ... The server "
+             "group may be terminated with ':<n>' to choose one of "
+             "multiple servers if there is more than one in the group. "
+             "For example: webworkers:0 will pick the first webworker.")
+
+    args = parser.parse_args()
+    if ':' in args.group:
+        group, index = args.group.rsplit(':', 1)
+        try:
+            index = int(index)
+        except (TypeError, ValueError):
+            parser.error("Non-numeric group index: {}".format(index))
+    else:
+        group = args.group
+        index = None
+
+    try:
+        servers = get_instance_group(args.environment, group)
+    except IOError as err:
+        parser.error(err)
+    except KeyError as err:
+        parser.error("Unknown group: {}\n".format(group))
+
+    if index is not None and index > len(servers) - 1:
+        sys.stderr.write(
+            "Invalid group index: {index}\n"
+            "Please specify a number between 0 and {max} inclusive\n"
+            .format(index=index, max=len(servers) - 1)
```

Comment:
@dannyroberts this is a slight change in behavior. Previously it would default to the last server in the group if the given index was too high. Now it's an error. I think it's better to be conservative here to prevent accidental login to the wrong server if someone had been using an ordinal and then the number of servers decreased and they just silently get dropped on a different server the next time they use that index. I'll be happy to change it back if you think this is too inconvenient.